WebRoot/js/zimbraMail/mail/view/prefs/ZmMailPrefsPage.js

```diff
@@ -834,40 +859,36 @@ function() {
 	this._numUsedText.innerHTML = AjxMessageFormat.format(ZmMsg.whiteBlackNumUsed, [this._listView.size(), this._max]);
 };
 
+ZmWhiteBlackList.prototype._SelectionListener = function (ev) {
+	var selection = this._listView.getSelection();
+	if (ev && ev.target && this._listView.delButtons[ev.target.id]) {
+		this._removeListener(ev);
+	}
+};
+
 ZmWhiteBlackList.prototype._setContent =
 function(templateId) {
 	this.getHtmlElement().innerHTML = AjxTemplate.expand("prefs.Pages#"+templateId, {id:this._htmlElId});
 
 	var id = this._htmlElId + "_EMAIL_ADDRESS";
 	var el = document.getElementById(id);
 	this._inputEl = new DwtInputField({parent:this, parentElement:id, size:35, hint:ZmMsg.enterEmailAddressOrDomain});
-	this._inputEl.getInputElement().style.width = "210px";
 	this._inputEl._showHint();
 	this._inputEl.addListener(DwtEvent.ONKEYUP, new AjxListener(this, this._handleKeyUp));
 	this.parent._addControlTabIndex(el, this._inputEl);
 
 	id = this._htmlElId + "_LISTVIEW";
-	el = document.getElementById(id);
-	this._listView = new DwtListView({parent:this, parentElement:id});
-	this._listView.addClassName("ZmWhiteBlackList");
+	el = Dwt.getElement(id);
+	this._listView = new ZmWhiteBlackListView({parent:this, parentElement:id, className:"ZmWhiteBlackList"});
+	this._listView.addSelectionListener(new AjxListener(this, this._SelectionListener));
 	this.parent._addControlTabIndex(el, this._listView);
 
 	id = this._htmlElId + "_ADD_BUTTON";
-	el = document.getElementById(id);
-	var addButton = new DwtButton({parent:this, parentElement:id});
-	addButton.setText(ZmMsg.add);
-	addButton.addSelectionListener(new AjxListener(this, this._addListener));
-	this.parent._addControlTabIndex(el, addButton);
-
-	id = this._htmlElId + "_REMOVE_BUTTON";
-	el = document.getElementById(id);
-	var removeButton = new DwtButton({parent:this, parentElement:id});
-	removeButton.setText(ZmMsg.remove);
-	removeButton.addSelectionListener(new AjxListener(this, this._removeListener));
-	this.parent._addControlTabIndex(el, removeButton);
-
+	el = Dwt.getElement(id);
+	el.innerHTML = AjxImg.getImageHtml("Add");
```

Comment:
Moved them to templates, now.
https://github.com/Zimbra/zm-web-client/commit/77231d0c726b0dcef692dae26f31fe56e5942136